Gems/ScriptAutomation/Code/Source/ScriptAutomationSystemComponent.cpp

```diff
@@ -17,49 +17,50 @@
 #include <AzCore/RTTI/BehaviorContext.h>
 #include <AzCore/Script/ScriptAsset.h>
 #include <AzCore/Script/ScriptContext.h>
+#include <AzCore/Script/ScriptSystemBus.h>
 #include <AzCore/Serialization/EditContext.h>
 #include <AzCore/Serialization/EditContextConstants.inl>
 #include <AzCore/Serialization/SerializeContext.h>
 
 #include <AzFramework/API/ApplicationAPI.h>
 #include <AzFramework/Asset/AssetSystemBus.h>
+#include <AzFramework/Script/ScriptComponent.h>
 
 namespace ScriptAutomation
 {
     namespace
     {
-        AZ::Data::Asset<AZ::ScriptAsset> LoadScriptAssetFromPath(const char* productPath)
+        AZ::Data::Asset<AZ::ScriptAsset> LoadScriptAssetFromPath(const char* productPath, AZ::ScriptContext& context)
         {
             AZ::IO::FixedMaxPath resolvedPath;
             AZ::IO::FileIOBase::GetInstance()->ResolvePath(resolvedPath, productPath);
 
-            AZ::Data::AssetId assetId;
-            AZ::Data::AssetCatalogRequestBus::BroadcastResult(
-                assetId,
-                &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath,
-                resolvedPath.c_str(),
-                AZ::AzTypeInfo<AZ::ScriptAsset>::Uuid(),
-                true
-            );
-
-            if (assetId.IsValid())
+            AZ::IO::FileIOStream inputStream;
+            if (inputStream.Open(resolvedPath.c_str(), AZ::IO::OpenMode::ModeRead))
             {
-                AZ::Data::Asset<AZ::ScriptAsset> asset = AZ::Data::AssetManager::Instance().GetAsset<AZ::ScriptAsset>(
-                    assetId, AZ::Data::AssetLoadBehavior::PreLoad
-                );
-                asset.BlockUntilLoadComplete();
+                AzFramework::ScriptCompileRequest compileRequest;
+                compileRequest.m_sourceFile = resolvedPath.c_str();
+                compileRequest.m_input = &inputStream;
 
-                if (!asset.IsReady())
+                auto outcome = AzFramework::CompileScript(compileRequest, context);
+                inputStream.Close();
+                if (outcome.IsSuccess())
                 {
-                    AZ_Assert(false, "Could not load '%s'", resolvedPath.c_str());
+                    AZ::Uuid id = AZ::Uuid::CreateRandom();
```

Comment:
@amzn-mike @AMZN-stankowi Is the actual UUID for the source asset based on the relative path from the product cache root that is lower cased?
If would the actual AssetId GUID be calculated by invoking AZ::Uuid::CreateName?
```suggestion
                    AZ::Uuid id = AZ::Uuid::CreateName(productPath);
```

Reply:
It appears so. `GetAssetIdByPath()`, which is what was being called earlier, ends up in [`GenerateAssetIdTEMP()`](https://github.com/aws-lumberyard-dev/o3de/blob/5e4e61c0da6e3bdc949a5733a4d94bd449696adf/Code/Framework/AzFramework/AzFramework/Asset/AssetCatalog.cpp#L484) if it doesn't already find the asset in the registry.

Reply:
Updated to create uuid based on path and then use `FindOrCreateAsset()` for cases where the asset already exists.